bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BFunctionType.java

```diff
@@ -133,6 +133,11 @@ public boolean equals(Object o) {
         if (!Arrays.equals(paramTypes, that.paramTypes)) {
             return false;
         }
+
+        if (retType == null) {
```

Comment:
AFAIK, the absence of a return type is always desuaged to be nil (and it is not never). The only exception I believe is with `function`, for which we don't have a return type, but we shouldn't get here due to the checks at L115 and L119. Can you share the sample for which this became null here?